src/sentry/notifications/utils/tasks.py

```diff
@@ -33,34 +63,21 @@ def async_send_notification(
     task_args = []
     for arg in args:
         if isinstance(arg, Model):
-            meta = type(arg)._meta
-            task_args.append(
-                {
-                    "type": "model",
-                    "app_label": meta.app_label,
-                    "model_name": meta.model_name,
-                    "pk": arg.pk,
-                    "key": None,
-                }
-            )
+            task_args.append(serialize_model(arg))
+        elif isinstance(arg, SimpleLazyObject):
+            task_args.append(serialize_lazy_object(arg))
         # maybe we need an explicit check if it's a primitive?
         else:
             task_args.append({"type": "other", "value": arg, "key": None})
     for key, val in kwargs.items():
         if isinstance(val, Model):
-            meta = type(val)._meta
-            task_args.append(
-                {
-                    "type": "model",
-                    "app_label": meta.app_label,
-                    "model_name": meta.model_name,
-                    "pk": val.pk,
-                    "key": key,
-                }
-            )
+            task_args.append(serialize_model(val, key))
+        elif isinstance(arg, SimpleLazyObject):
+            task_args.append(serialize_lazy_object(arg, key))
```

Comment:
This looks like the change we'll want to make. We can't change the task producers just yet though as the tasks they emit could be processed by old-code-workers that aren't able to process tasks from new code.

What we'll likely need to do instead, is change the task body so it can handle both the current and 'next' parameters set. Then we can follow that up with a change that removes submitting the ORM object to this task, and finally another change to remove the compatibility shims.